src/overlay/Peer.cpp

```diff
@@ -1061,6 +1061,11 @@ Peer::recvAuth(StellarMessage const& msg)
     {
         CLOG(WARNING, "Overlay") << "New peer rejected, all slots taken";
         mDropInRecvAuthRejectMeter.Mark();
```

Comment:
wouldn't it be better to just move this code (if condition etc) to right before `// send SCP State` ?
This would avoid duplicating the handshaking code (and in this case - the handshake was successful we're dropping the peer for another reason).